core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -391,6 +412,46 @@ void setupVersionMetrics() {
              .register(meterRegistry);
     }
 
+    /**
+     * Sets up gauge metric for each server certificate.
+     */
+    private void setupTlsMetrics(SslContext sslContext, String hostname) {
+        final MeterRegistry meterRegistry = config().meterRegistry();
+
+        final SSLSession sslSession = validateSslContext(sslContext);
+        for (Certificate certificate : sslSession.getLocalCertificates()) {
+            try {
+                final X509Certificate x509Certificate = (X509Certificate) certificate;
```

Comment:
Could we `continue` if `certificate` is not `instanceof X509Certificate`?

Reply:
I'm not sure if I understand your question correctly😅. Is what you mean is that we should skip setting up metric if it's not instance of `X509Certificate` as below, right?

```java
if (!(certificate instanceof X509Certificate)) {
    continue;
}
final X509Certificate x509Certificate = (X509Certificate) certificate;
final String commonName = getCommonName(x509Certificate);
```

Reply:
> What do you mean is that we should skip setting up metric.

+1 It is.

Reply:
Correct :+1:

Reply:
Pushed a commit that adds the instanceof check to your branch. 🙂

Reply:
Thank you, @trustin. 🙏